src/middlewared/middlewared/plugins/tunables.py

```diff
@@ -28,25 +28,40 @@ class Config:
         datastore_prefix = 'tun_'
         cli_namespace = 'system.tunable'
 
-    SYSTEM_TUNABLES = deque()
+    SYSCTLS = set()
 
     ENTRY = Patch(
         'tunable_create', 'tunable_entry',
         ('add', Int('id')),
     )
 
     @private
-    def get_system_tunables(self):
-        if not TunableService.SYSTEM_TUNABLES:
+    def get_sysctls(self):
+        if not TunableService.SYSCTLS:
             tunables = subprocess.run(['sysctl', '-aN'], stdout=subprocess.PIPE)
             for tunable in filter(lambda x: x, tunables.stdout.decode().split('\n')):
-                TunableService.SYSTEM_TUNABLES.append(tunable)
-        return TunableService.SYSTEM_TUNABLES
+                TunableService.SYSCTLS.add(tunable)
+        return TunableService.SYSCTLS
+
+    @private
+    def get_sysctl(self, var):
+        with open(f'/proc/sys/{var.replace(".", "/")}', 'r') as f:
+            return f.read().strip()
+
+    @private
+    def set_sysctl(self, var, value):
+        with open(f'/proc/sys/{var.replace(".", "/")}', 'w') as f:
+            f.write(value)
+
+    @private
+    def reset_sysctl(self, tunable):
+        self.set_sysctl(tunable['var'], tunable['orig_value'])
 
     @private
-    def get_or_set(self, var, value=None):
-        with open(f'/proc/sys/{var.replace(".", "/")}', 'r' if not value else 'w') as f:
-            return f.read().strip() if not value else f.write(value)
+    async def handle_tunable_change(self, tunable):
+        if tunable['type'] == 'UDEV':
+            await self.middleware.call('etc.generate', 'udev')
+            await run(['udevadm', 'control', '--reload-rules'])
```

Comment:
Does this do anything?

Reply:
What exactly do you suspect of not doing anything?

Reply:
What do you expect it to do?

Reply:
Reload the rules explicitly. Do you also want `udevadm trigger`?

Reply:
I actually don't even see --reload-rules in Debian man pages. Also, if I'm understanding this it reloads all rules. "Signal systemd-udevd to reload the rules files and other databases like the kernel module index. Reloading rules and databases does not apply any changes to already existing devices; the new configuration will only be applied to new events."

Reply:
@yocalebo `--reload-rules` is an alias for `-R` https://github.com/systemd/systemd/blob/main/src/udev/udevadm-control.c#L66 I don't know why is it in all examples and in my fingers memory, but I can change it to `-R` if you want.

Reply:
I don't think it would be good to run `trigger` on every tunable change. Users can do that themselves if they want.

Reply:
I don't believe udevadm control reload does anything but man pages are horrible in that regard and I'm too lazy to go read source right now.